dotnet/src/Agents/Core/Chat/DefaultTerminationStrategy.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// The termination strategy attached to the default state of <see cref="ChatExecutionSettings.TerminationStrategy"/>.
+/// Terminates immediate, by default.  Behavior can be overriden via <see cref="DefaultTerminationStrategy.DisableTermination"/>.
+/// </summary>
+public sealed class DefaultTerminationStrategy : TerminationStrategy
+{
+    /// <summary>
+    /// Strategy terminates by default, but may be overridden by setting this property.
+    /// The <see cref="TerminationStrategy.MaximumIterations"/> property provides additional
+    /// control to this strategy.
+    /// </summary>
+    public bool DisableTermination { get; set; }
+
+    /// <inheritdoc/>
+    protected override Task<bool> ShouldAgentTerminateAsync(Agent agent, IReadOnlyList<ChatMessageContent> history, CancellationToken cancellationToken = default)
+    {
+        bool shouldTerminate = !this.DisableTermination;
```

Comment:
Is this line necessary?

Reply:
Its just for readability / clarify.  Sometimes double negatives can be non-obvious (which is why they are discouraged in grammar).  I don't believe there's any runtime impact.  The compiler can optimize this expression.

Reply:
This line can be turned into a comment, or the variable can be used in the next line: `return Task.FromResult(shouldTerminate);`  
I wonder why the compiler does not detect the assigned but unused variable. Does this project have different code analysis rules?

Reply:
If you feel strongly on this, no problem.

Reply:
In rechecking, I believe this code has already been entirely removed.